magik-language-server/src/main/java/nl/ramsolutions/sw/magik/languageserver/MagikTextDocumentService.java

```diff
@@ -631,21 +632,20 @@ public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completio
         });
   }
 
-  @Override
-  public CompletableFuture<List<? extends TextEdit>> formatting(
-      final DocumentFormattingParams params) {
+  private CompletableFuture<List<? extends TextEdit>> handleFormatting(
```

Comment:
Please keep the duplicate code, lets keep this as simple (and consistent) as possible.

If there should be any de-duplication, perhaps the timing code can be simplified and re-used everywhere.